File: Backend.Service/Entities/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Backend.Service.Consts;
using Backend.Service.Entities.Poco;
using Microsoft.EntityFrameworkCore;
using NpgsqlTypes;

namespace Backend.Service.Entities
{
    [Table("Products")]
    [Index(nameof(ProductCode), IsUnique = true)]
    public class Product : BaseEntity
    {
        public string Name { get; set; } = null!;
        public Guid ProductCode { get; set; } = Guid.NewGuid();

        [Column(TypeName = "jsonb")]
        public ICollection<Media> Medias { get; set; } = new List<Media>();

        [Column(TypeName = "text")]
        public string? Description { get; set; } = string.Empty;

        [Column(TypeName = "text")]
        public string? ShortDescription { get; set; } = string.Empty;

        public double Price { get; set; } = 0.0;
        public int Quantity { get; set; } = 0;
        public int ImportQuantity { get; set; } = 0;

        // Đã thử bỏ cái này vào, vô dụng
        //[Required]
        //[MaxLength(25)]
        public ProductStatus Status { get; set; } = ProductStatus.Available;

        public int CategoryId { get; set; }
        public Category Category { get; set; } = null!;

        // FTS
        public NpgsqlTsVector SearchVector { get; set; }

        #region One to many relationships

        public virtual ICollection<OrderDetail> OrderDetails { get; set; }

        public virtual ICollection<CartItem> CartItems { get; set; }

        public virtual ICollection<Feedback> Feedbacks { get; set; }
        #endregion
    }
}

File: frontend.product/src/models/Product.js
class Product {
  name;
  productCode;
  medias;
  description;
  price;
  quantity;
  importQuantity;
  categoryId;
  categoryName;
  status;
  categoryType;
  id;
  createdDate;
  updatedDate;
  updatedBy;
  addedBy;
  isDeleted;

  constructor(
    name,
    productCode,
    medias,
    description,
    price,
    quantity,
    importQuantity,
    categoryId,
    categoryName,
    status,
    categoryType,
    id,
    createdDate,
    updatedDate,
    updatedBy,
    addedBy,
    isDeleted
  ) {
    this.name = name;
    this.productCode = productCode;
    this.medias = medias;
    this.description = description;
    this.price = price;
    this.quantity = quantity;
    this.importQuantity = importQuantity;
    this.categoryId = categoryId;
    this.categoryName = categoryName;
    this.status = status;
    this.categoryType = categoryType;
    this.id = id;
    this.createdDate = createdDate;
    this.updatedDate = updatedDate;
    this.updatedBy = updatedBy;
    this.addedBy = addedBy;
    this.isDeleted = isDeleted;
  }
}
export default Product;
